src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -32,56 +32,82 @@ namespace Microsoft.Diagnostics.Tracing.Analysis
     /// </summary>
     public static class TraceLoadedDotNetRuntimeExtensions
     {
+        internal static readonly string LoadedDotNetRuntimesKey = "Computers/LoadedDotNetRuntimes";
+
         public static void NeedLoadedDotNetRuntimes(this TraceEventDispatcher source)
         {
+            if (source.userData.ContainsKey(LoadedDotNetRuntimesKey))
+            {
+                // The callbacks are already registered for this source
+                return;
+            }
+
             // ensure there are base processes
```

Comment:
@jeffschwMSFT I went ahead and modified the new implementation to behave the same as the old implementation, except without the race condition for concurrent use. The one part that doesn't make sense to me right now is why you'd want to reset the collected information. Do you have examples of this case?

The scenario that still makes me uncomfortable (even though it's the same as the previous implementation) is the following:

1. Call `NeedProcesses` to track process information
2. Wait some amount of time
3. Call `NeedLoadedDotNetRuntimes` to start tracking .NET runtime information
  * ➡️ Information in `TraceProcesses` is reset